nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -128,12 +135,25 @@ def init(cls: Type[T], name: str) -> Type[T]:
                 imap_port=url_components.port or 993,
             )
         if scheme == "https" and url_components.netloc.split(":")[0] == "accounts.google.com":
-            return GmailAPIServiceAccount(
-                name=name,
-                url=url,
-                account=config.get("account"),
-                credentials_file=config.get("credentials_file"),
-            )
+            creds_filename = config.get("credentials_file")
+            if not creds_filename:
+                raise ValueError(f"Credentials_file for {name} not found in PLUGINS_CONFIG.")
+
+            if not os.path.isfile(creds_filename):
+                raise ValueError(f"Credentials_file {creds_filename} for {name} is not available.")
+
+            with open(creds_filename) as crededentials_file:
+                credentials = json.load(crededentials_file)
+                if credentials.get("type") == "service_account":
+                    gmail_api_class = GmailAPIServiceAccount
+                else:
+                    gmail_api_class = GmailAPIOauth
```

Comment:
Perhaps this could be an `else if` to check specifically for OAuth2 type, and `else` raise an exception for an unknown/unsupported type?